src/course-updates/hooks.jsx

```diff
@@ -0,0 +1,114 @@
+import { useDispatch, useSelector } from 'react-redux';
+import moment from 'moment/moment';
+import { useEffect, useState } from 'react';
+import { useToggle } from '@edx/paragon';
+
+import { getCourseHandouts, getCourseUpdates } from './data/selectors';
+import { REQUEST_TYPES } from './constants';
+import { COMMA_SEPARATED_DATE_FORMAT } from '../constants';
+import {
+  createCourseUpdateQuery,
+  deleteCourseUpdateQuery,
+  editCourseHandoutsQuery,
+  editCourseUpdateQuery,
+  fetchCourseHandoutsQuery,
+  fetchCourseUpdatesQuery,
+} from './data/thunk';
+
+const useCourseUpdates = ({ courseId }) => {
+  const dispatch = useDispatch();
+  const initialUpdate = { id: 0, date: moment().utc().toDate(), content: '' };
+
+  const [requestType, setRequestType] = useState('');
+  const [isUpdateFormOpen, openUpdateForm, closeUpdateForm] = useToggle(false);
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
+  const [currentUpdate, setCurrentUpdate] = useState(initialUpdate);
+
+  const courseUpdates = useSelector(getCourseUpdates);
+  const courseHandouts = useSelector(getCourseHandouts);
+
+  const courseUpdatesInitialValues = requestType === REQUEST_TYPES.edit_handouts
+    ? courseHandouts
+    : currentUpdate;
+
+  const handleOpenUpdateForm = (type, courseUpdate) => {
+    setRequestType(type);
+
+    switch (type) {
+    case REQUEST_TYPES.add_new_update:
+      setCurrentUpdate(initialUpdate);
+      break;
+    case REQUEST_TYPES.edit_update:
+      setCurrentUpdate(courseUpdate);
+      break;
+    default:
+      window.scrollTo(0, 0);
+    }
+
+    openUpdateForm();
+  };
+
+  const handleOpenDeleteUpdateForm = (courseUpdate) => {
```

Comment:
I think it's best to rename this function to a name that more accurately describes what it does. A similar suggestion for the second parameter of the `useToggle` `openDeleteModal` hook. Can we name this functionality more clearly?